### DISCUSSION TITLE
What Are the Minimum Requirements for OpenTrackIO Compliance?

### DISCUSSION THREAD

**Discussion by flow-dev** [SMPTE/ris-osvp-metadata-camdkit#126] [Q&A] [answered]
# What Are the Minimum Requirements for OpenTrackIO Compliance?

Hello,

From reading the [OpenTrackIO Documentation v0.9.0](https://www.opentrackio.org/), it describes a structure designed for Real-time Transport Protocol (RTP) and ST2110-41, along with capabilities such as jumbo packet splitting and SDP signaling links. However, what are the minimum implementation requirements for OpenTrackIO compliance?

When starting OpenTrackIO compliance, implementing everything from the beginning—such as a structure designed for RTP, ST2110-41, jumbo packet splitting, and SDP signaling links—can be overwhelming. I would like to understand the minimum implementation requirements for OpenTrackIO compliance. I would like to start by addressing the areas where the 29-byte FreeD format was clearly insufficient.

Based on the documentation, here is what I have interpreted as the minimum implementation requirements. I would appreciate your guidance and clarification.

1. **UDP Multicast (IPv4)**: Multicast addressing is recommended, but Unicast is also acceptable (similar to FreeD's Unicast).

2. **No Jumbo Packet Support**: Data transmission with packets that fit within 1460 bytes (MTU-compliant) is acceptable.

3. **Minimum Data Content**:  
   The *Recommended minimum VP example* ([example link](https://www.opentrackio.org/examples/recommended_dynamic_example.json)) is indicated as the minimum structure. However, to meet the minimum requirements for camera and lens geometric parameters, should the *Recommended minimum VP example with static data* ([example link](https://www.opentrackio.org/examples/recommended_static_example.json))—which includes static information like `activeSensorPhysicalDimensions`—be considered the minimum required structure?

4. **Static Data Transmission**:  
   If static information is included as in the *Recommended minimum VP example with static data*, can static data be combined with dynamic data and sent together (instead of transmitting static data every 2 seconds), to simplify data transmission within the 1460-byte MTU limit?

5. **CBOR Preference**:  
   It seems CBOR is recommended because it achieves better data size efficiency compared to JSON. If CBOR is used, is there a need for special headers or mechanisms to distinguish between JSON and CBOR formats?

6. **Data Type Definitions**:  
   Regarding the *Recommended minimum VP example*, it seems that not all data type definitions for JSON fields are strictly specified. Are there guidelines available? For instance, are there recommendations on the number of significant digits for float values? JSON allows an arbitrary number of digits, so having guidelines on data types and significant figures for each field would be helpful.  
   ([Schema link](https://www.opentrackio.org/schema.json))

---

When illustrated, I am assuming a structure like this. Does this align with the OpenTrackIO guidelines?
![image](https://github.com/user-attachments/assets/d593dff3-e72d-4580-ad44-856cf0440ccb)

Thank you for your response and guidance.

**Comment by jamesmosys** [comment]
Thanks @flow-dev ! We are confirming on 5/12 that GitHub discussions are the correct place for us to handle these kinds of support questions. Once confirmed we will get back to you on your specific questions. Thanks for being the first discussions contributor!

**Comment by jamesmosys** [comment, marked as answer]
Hi @flow-dev, Thanks for you query - there is a [PR under review](https://github.com/SMPTE/ris-osvp-metadata-camdkit/pull/134) right now that we hope will answer your questions. In short yes there will be a minimal UDP version with a simple header that helps define the content of the packet. While RTP and ultimately 2110 are recommended, 'raw' UDP is also supported. The new docs will also describe how to send multiple 1460-byte payloads, and I will add some improved description on floating point precision. I'll leave this discussion open until the PR is merged and we have confirmed it covers everything you have raised. Thanks!

**Reply by jamesmosys** [reply]
This has now been merged so I'm closing this discussion. Thanks!

**Reply by flow-dev** [reply]
@jamesmosys 
Has OpenTrackIO version **1.0.0** finally been released?

**Reply by jamesmosys** [reply]
Final stages now. Looking like Monday :)
